client/src/lib/api/files.ts

```diff
@@ -32,6 +43,36 @@ export async function getSignedUploadURLs(user: User, fileNames: String[]) {
   return new Map<string, string>(Object.entries(result));
 }
 
+/**
+ * Get all files from the given dataset for the given user
+ *
+ * @param user User whose files need to be retrieved
+ * @returns A Promise that resolves to a list of UserFile objects
+ */
+export async function getFilesFromDataset(
+  user: User,
+  datasetName: string
+): Promise<UserFile[]> {
+  const getDataset = async () => {
+    const docRef = doc(firestore, `users/${user!.uid}/datasets/${datasetName}`);
+    const docSnapShot = await getDoc(docRef);
+    const returnedDocument: DocumentData = docSnapShot.data() as DocumentData;
+    return returnedDocument as Dataset;
+  };
+  console.log(user);
+  console.log(datasetName);
+  const dataset: Dataset = await getDataset();
+  console.log(dataset);
+
+  const returnedFiles = new Array<UserFile>();
+  for (const filename of dataset.files) {
+    const docRef = doc(firestore, `users/${user!.uid}/files/${filename}`);
```

Comment:
I'd make this a separate function called `getFile` within this file. Also I guess we don't really need to know the extra file information and the filename would be enough, but this is fine for now, just something I guess we keep in mind for the future, if we're worried about making tonnes of consecutive calls with big datasets.